### PR TITLE
Added API v1 compute resource actions and tests

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: test/functional/api/v1/compute_resources_controller_test.rb [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 class Api::V1::ComputeResourcesControllerTest < ActionController::TestCase
 
-  valid_attrs = { :name => 'special_compute', :provider => 'EC2', :url => 'eu-west1', :user => 'user@example.com', :password => 'secret' }
+  setup do
+   Fog.mock!
```
Comment (ohadlevy) [inline test/functional/api/v1/compute_resources_controller_test.rb]: this breaks other tests i guess which do not relay on fog mock, can you disable it in  a teardown and validate all tests pass?

**Comment by isratrade** [comment]
@ohadlevy, I added Fog.mock! to teardown and all tests pass now.

File: test/functional/api/v1/compute_resources_controller_test.rb [hunk 2]
```diff
@@ -2,7 +2,15 @@
...
-  valid_attrs = { :name => 'special_compute', :provider => 'EC2', :url => 'eu-west1', :user => 'user@example.com', :password => 'secret' }
+  def setup
+    Fog.mock!
+  end
+
+  def teardown
+    Fog.unmock!
+  end
+
+  valid_attrs = { :name => 'special_compute', :provider => 'EC2', :region => 'eu-west-1', :user => 'user@example.com', :password => 'secret' }
```
Comment (ohadlevy) [inline test/functional/api/v1/compute_resources_controller_test.rb:13]: not a big deal, but i think it makes more sense to be a method vs a class variable

**Comment by ohadlevy** [comment]
merged - thanks!

**Pull request closed without merging**